tests/integration/triggering.py

```diff
@@ -14,9 +13,30 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#-------------------------------------------------------------------------------
-# Test fail triggering
-. "$(dirname "$0")/test_header"
-set_test_number 2
-reftest
-exit
+
+async def test_fail(flow, scheduler, run, reflog, complete, validate):
+    id_ = flow({
+        'scheduler': {
+            'allow implicit tasks': 'true'
+        },
+        'scheduling': {
+            'graph': {
+                'R1': 'foo:failed => bar'
+            }
+        },
+        'runtime': {
```

Comment:
```suggestion
        'runtime': {
            'root': {
                'simulation': {'default run length': 'PT0S'}
            },
```
Makes this test run in 2-3s! It might be worth coming up with a fixture that can automatically merge this into the flow config dict?

Reply:
Although https://github.com/cylc/cylc-flow/blob/ec76e2e1a7069f52a5fcc05eef3bb3f4641224e0/tests/integration/test_examples.py#L234

runs in under 3s too without any special config... I'm not sure how that works

Reply:
https://cylc.github.io/cylc-doc/stable/html/reference/config/workflow.html#flow.cylc[runtime][%3Cnamespace%3E][simulation]?

Reply:
Another look suggests that it's the implicit tasks - I tried taking out all the task definition and got the speedup, although of course the task didn't fail when I wanted it to for the test.

Reply:
Ah I've got it, the integration tests already run in simulation mode with a default run length of `PT0S`. But it doesn't intelligently merge the config dict so this is lost when you specify the runtime section

Reply:
https://github.com/wxtim/cylc/pull/58

Reply:
Thanks!